src/components/autocomplete/js/autocompleteController.js

```diff
@@ -64,6 +66,18 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
     $mdUtil.initOptionalProperties($scope, $attrs, { searchText: null, selectedItem: null });
     $mdTheming($element);
     configureWatchers();
+
+    if ($scope.componentId) {
+      removeFromRegistry = $mdComponentRegistry.register({
+        // Restricted API interface for the component registry.
+        // We don't want to provide the whole controller as API.
+        clearCache: function() {
+          // Run the clearCache method in the controller context.
+          ctrl.clearCache();
```

Comment:
@EladBezalel Please review again - and about this specific line:
-  It is needed to provide a new object with all the needed functions, which should accessable in the component instance.

Also I think, you're wondering about the extra `clearCache` method in the controller, but I think it's good to implement this as a new function in the controller. But if you have other thoughts about the `clearCache` controller function, let me know - I'm up to move this in-line.